docs(tutorial): Irving tutorial

jameswburke opened the pull request:
## Summary
This is the Irving tutorial. It's still a bit of a work in progress, but we're ready for feedback.

## Notes for reviewers
You can see this deployed at https://storybook-qa.irvingjs.com/?path=/docs/tutorial-introduction--page

## Changelog entries
* **Added**: Tutorial section.

## Ticket(s)
https://alleyinteractive.atlassian.net/browse/IRV-892

jameswburke commented on stories/01-tutorial/05-Register-the-Component-in-WordPress.stories.mdx:
```diff
@@ -0,0 +1,154 @@
+<img src="/assets/img/tutorial/04/finished.png" style={{maxWidth: "600px", margin: "auto", width: "100%", display: "flex"}} />
+
+## That's It!
+Now that you've completed the tutorial, you're reach to conquer Irving!
```
```suggestion
Now that you've completed the tutorial, you're ready to conquer Irving!
```

joemcgill commented on stories/01-tutorial/05-Register-the-Component-in-WordPress.stories.mdx:
```diff
@@ -0,0 +1,154 @@
+
+<img src="/assets/img/tutorial/04/finished.png" style={{maxWidth: "600px", margin: "auto", width: "100%", display: "flex"}} />
+
+## That's It!
```
Seems like we need one more step here that explains the config/child callbacks, otherwise there really isn't much value in passing static data from the CMS to the front end.

jameswburke replied:
You could use a component like this in multiple places, and the static data from the layout could be modified depending on the context, so I wanted to demonstrate this.

Agreed we need to go into config & child callbacks, but I need to draw the line somewhere to get this merged. It'll be added in a followup PR for sure.

joemcgill reviewed (commented):
This seems really well put together to me and breaks down the main steps to getting a project up and running well. I left a few notes, but am really thrilled how this is put together. Nice work!

jameswburke commented on stories/01-tutorial/01-Setting-up-WordPress.stories.mdx:
````diff
@@ -0,0 +1,110 @@
+```
+
+## Update `API_ROOT_URL` in `.env`
+Now that we've confirmed WP Irving is setup and ready, replace `API_ROOT_URL` in your `.env`,
````
That's all in the [Quick Start](https://storybook.irvingjs.com/?path=/docs/intro-quick-start--page) - which is flagged as a prerequisite before starting the tutorial.

jameswburke commented:
@joemcgill & @lmfern, I've addressed feedback from both of you, and also cleaned up some of the later sections of the tutorial.

Ready for another look.

joemcgill reviewed (approved):
👍🏻
